spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java

```diff
@@ -88,10 +102,48 @@ public DispatchQueryResponse submit(
             dispatchQueryRequest.getDatasource(),
             System.currentTimeMillis() - startTime,
             System.currentTimeMillis());
-    String resultIndex = dataSourceMetadata.getResultIndex();
-    createIndexDMLResult(stateStore, resultIndex).apply(indexDMLResult);
+    createIndexDMLResult(stateStore, dataSourceMetadata.getResultIndex()).apply(indexDMLResult);
+    return asyncQueryId;
+  }
 
-    return new DispatchQueryResponse(asyncQueryId, DROP_INDEX_JOB_ID, resultIndex, null);
+  private void executeIndexOp(
```

Comment:
I'm adding `VACUUM` operation and thinking we may want to make this abstract and have separate handler for Drop, Alter and Vacuum in future?

Reply:
Yes, I was thinking to abstract out and then the changes are becoming huge. Considering the timeline and since the we have only two use cases I just didn't refactor.  But makes sense to abstract out these operations individually and bring common pieces to an abstract class.